### PR TITLE
Python: Improve pyyaml support

### PR DESCRIPTION
**Pull request by RasmusWL**

### CODE REVIEW

**Review by yoff — DISMISSED** [review]
LGTM

File: python/change-notes/2021-02-10-yaml-more-loading-functions.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+lgtm,codescanning
+* Improved modeling for the `PyYAML` PyPI package (imported as `yaml`), now supporting `safe_load`, `unsafe_load`, and `full_load` (as well as the `..._load_all` functions). In the current version of PyYAML (5.4.1), only `safe_load` is known to not be safe from code execution exploits. Calls to the other functions are now modeled as sinks of the _Deserializing untrusted input_ (`py/unsafe-deserialization`) query.
```
Comment (felicitymay) [inline python/change-notes/2021-02-10-yaml-more-loading-functions.md:2]: Would this change work, to simplify "known to not be safe from":

```suggestion
* Improved modeling for the `PyYAML` PyPI package (imported as `yaml`), now supporting `safe_load`, `unsafe_load`, and `full_load` (as well as the `..._load_all` functions). In the current version of PyYAML (5.4.1), only `safe_load` is known to be vulnerable to code execution exploits. Calls to the other functions are now modeled as sinks of the _Deserializing untrusted input_ (`py/unsafe-deserialization`) query.
```

**Review by felicitymay — COMMENTED** [review]
Once small text suggestion - otherwise LGTM to me.

File: python/change-notes/2021-02-10-yaml-more-loading-functions.md (hunk 1 above)
Comment (RasmusWL) [reply python/change-notes/2021-02-10-yaml-more-loading-functions.md]: Thanks, I must have rewritten that once too many, and ended up absolutely botching it :laughing: The only one that is safe is `safe_load`, which is hopefully what the change note also says now :laughing:

File: python/change-notes/2021-02-10-yaml-more-loading-functions.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+lgtm,codescanning
+* Improved modeling for the `PyYAML` PyPI package (imported as `yaml`), now supporting `safe_load`, `unsafe_load`, and `full_load` (as well as the `..._load_all` functions). In the current version of PyYAML (5.4.1), only `safe_load` is known to be safe from code execution exploits. Calls to the other functions are now modeled as sinks of the _Deserializing untrusted input_ (`py/unsafe-deserialization`) query.
```
Comment (felicitymay) [inline python/change-notes/2021-02-10-yaml-more-loading-functions.md:2]: Suggest:

```suggestion
* Improved modeling of the `PyYAML` PyPI package (imported as `yaml`) now includes `safe_load`, `unsafe_load`, and `full_load` (as well as the `..._load_all` functions). In the current version of PyYAML (5.4.1), only `safe_load` is known to be safe from code execution exploits. Consequently, calls to the other functions are modeled as sinks of the _Deserializing untrusted input_ (`py/unsafe-deserialization`) query.
```

**Review by felicitymay — COMMENTED** [review]
Small suggestion for the change notes. Otherwise the text changes LGTM.

File: python/change-notes/2021-02-10-yaml-more-loading-functions.md (hunk 2 above)
Comment (RasmusWL) [reply python/change-notes/2021-02-10-yaml-more-loading-functions.md]: Thanks :+1:

File: python/change-notes/2021-02-10-yaml-more-loading-functions.md (hunk 1 above)
Comment (felicitymay) [reply python/change-notes/2021-02-10-yaml-more-loading-functions.md]: Thanks for the update. Clearly I should also have been concentrating harder when I reviewed too.

Content all LGTM now 👍🏻

File: python/change-notes/2021-02-10-yaml-more-loading-functions.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+lgtm,codescanning
+* Improved modeling of the `PyYAML` PyPI package (imported as `yaml`) now includes `safe_load`, `unsafe_load`, and `full_load` (as well as the `..._load_all` functions). In the current version of PyYAML (5.4.1), only `safe_load` is known to be safe from code execution exploits. Consequently, calls to the other functions are modeled as sinks of the _Deserializing untrusted input_ (`py/unsafe-deserialization`) query.
```
Comment (yoff) [inline python/change-notes/2021-02-10-yaml-more-loading-functions.md:2]:
```suggestion
lgtm,codescanning
* Improved modeling of the `PyYAML` PyPI package (imported as `yaml`) now includes `safe_load`, `unsafe_load`, and `full_load` (as well as the `..._load_all` functions). In the current version of PyYAML (5.4.1), only `safe_load` and `safe_load_all` are known to be safe from code execution exploits. Consequently, calls to the other functions are modeled as sinks of the _Deserializing untrusted input_ (`py/unsafe-deserialization`) query.
```

**Review by yoff — CHANGES_REQUESTED** [review]
We might as well be completely precise and un-misunderstandable here..

File: python/change-notes/2021-02-10-yaml-more-loading-functions.md (hunk 3 above)
Comment (RasmusWL) [reply python/change-notes/2021-02-10-yaml-more-loading-functions.md]: sure thign

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**